### PR TITLE
Add source/binary compat to BC template

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/dotnet-breaking-change.md [hunk 1]
```diff
@@ -32,6 +32,13 @@ assignees: ''
 
 ### New behavior
 
+### Category
+
+<!-- Choose one of the following compatibility categories: -->
+
+- [ ] Binary compatibility (the newer version can be used without recompiling)
+- [ ] Source compatibility (successfully recompiling against newer version requires source changes)
```
Comment (tdykstra) [inline .github/ISSUE_TEMPLATE/dotnet-breaking-change.md:40]: "requires source changes" sounds like **in**compatibility (?)

**Review by tdykstra — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/dotnet-breaking-change.md (hunk 1 above)
Comment (gewarren) [reply .github/ISSUE_TEMPLATE/dotnet-breaking-change.md:40]: The idea is that the breaking change affects source compatibility. Does that make sense?

Comment (tdykstra) [reply .github/ISSUE_TEMPLATE/dotnet-breaking-change.md:40]: Yes, it's OK as-is, it just seems out of sync with the explanatory phrase for binary compatibility, i.e., one identifies compatibility and one identifies incompatibility.

Comment (gewarren) [reply .github/ISSUE_TEMPLATE/dotnet-breaking-change.md]: I see. I've made them consistent now, thanks.

**Pull request merged**